### PR TITLE
Formal code cleaning

### PR DESCRIPTION
**Pull request by phako**
- Disable deprecation warnings
- Fix linter findings

### CODE REVIEW

File: include/FeedingFanOut.h [hunk 1]
```diff
@@ -96,7 +99,7 @@ namespace ChimeraTK {
     // Add the consuming accessors
     // TODO FanOut constructors and addSlave should get refactoring
     for(auto el : consumerImplementationPairs) {
-      addSlave(el.first, el.second);
+      addSlaveImpl(el.first, el.second);
```
Comment (mhier) [inline include/FeedingFanOut.h:101]: Wouldn't it be easier yet still sufficient to call it explicitly by ```FeedingFanOut<UserType>::addSlave```?

**Review by mhier — APPROVED** [review]

**Comment by mhier** [comment]
Wait: `ArrayAccessor<UserType>::writeDestructively()` still contains a reference to `node` without underscore, probably it is never referenced inside the ApplicationCore project itself.

**Review by mhier — CHANGES_REQUESTED** [review]
Compilation of llrfctrl server fails

File: include/Module.h [hunk 2]
```diff
@@ -32,7 +32,7 @@ namespace ChimeraTK {
      * to allow constructor inheritance of modules owning other modules. This
      * constructor will not actually be called then. See this bug report:
      * https://gcc.gnu.org/bugzilla/show_bug.cgi?id=67054 */
-    Module() : _owner{nullptr} {}
+    Module() {}
```
[the remark this answers is not in the document]
Comment (phako) [reply include/Module.h]: Yeah, looks like sometimes you need multiple rounds of linting

File: include/DeviceManager.h [hunk 3]
```diff
@@ -22,7 +22,7 @@ namespace ChimeraTK {
     /**
      * Create DeviceManager which handles device exceptions and performs the recovery.
      */
-    DeviceManager(Application* application, const std::string& deviceAliasOrCDD);
+    DeviceManager(Application* application, const std::string& deviceAliasOrURI);
```
[the remark this answers is not in the document]
Comment (phako) [reply include/DeviceManager.h:25]: I think the linter fix prefers the name from the implementation

[the remark this answers is not in the document]
Comment (phako) [reply include/DeviceManager.h:25]: Btw, there is also getDeviceAliasOrUri() api in the module and the manager.

[the remark this answers is not in the document]
Comment (mhier) [reply include/DeviceManager.h:25]: Hm, ok. Probably all those names would need to be changed eventually :-) I would start with changing the argument name here into `deviceAliasOrCDD` in both definition and implementation, as this is no API change.

File: include/RecoveryHelper.h [hunk 4]
```diff
@@ -18,7 +20,7 @@ namespace ChimeraTK {
 
     explicit RecoveryHelper(
         boost::shared_ptr<TransferElement> a, VersionNumber v = VersionNumber(nullptr), uint64_t order = 0)
-    : accessor(a), versionNumber(v), writeOrder(order) {}
+    : accessor(std::move(std::move(a))), versionNumber(v), writeOrder(order) {}
```
[the remark this answers is not in the document]
Comment (phako) [reply include/RecoveryHelper.h:23]: Fixed. run-clang-tidy seems to have some minor problems with merging fixes properly, there was one line with 8 [[nodiscard]] annotations

File: include/FeedingFanOut.h (hunk 1 above)
Comment (phako) [reply include/FeedingFanOut.h:101]: seems it is, yes

**Comment by phako** [comment]
```
jgeorg@jgeorg-ubuntu2204:~/workflow-scripts/build/llrfctrl-debug$ ctest -j 55
Test project /home/jgeorg/workflow-scripts/build/llrfctrl-debug
      Start  1: coding_style
      Start  2: llrfctrl-xmlGenerator
      Start  3: test_llrfctrl
      Start  4: testAdcChannel
      Start  5: testAmplitudePhaseError
      Start  6: testAutomation
      Start  7: testBeamLoadingCompensation
      Start  8: testControllerCoefficients
      Start  9: testDaqTriggerMux
      Start 10: testFirmwareSEUDetection
      Start 11: testLearningFeedForward
      Start 12: testOVC
      Start 13: testPhaseSetter
      Start 14: testReferencePhaseMonitor
      Start 15: testReferencePhaseMonitor_slave
      Start 16: testRotationAndScalingTable
      Start 17: testTableGeneration
      Start 18: testTriggerIntervalMonitor
      Start 19: testVMOffsetCalibration
 1/19 Test  #9: testDaqTriggerMux .................   Passed    1.04 sec
 2/19 Test  #1: coding_style ......................   Passed    3.83 sec
 3/19 Test  #5: testAmplitudePhaseError ...........   Passed   18.95 sec
 4/19 Test  #2: llrfctrl-xmlGenerator .............   Passed   35.08 sec
 5/19 Test #16: testRotationAndScalingTable .......   Passed   60.86 sec
 6/19 Test #19: testVMOffsetCalibration ...........   Passed   69.72 sec
 7/19 Test #13: testPhaseSetter ...................   Passed   76.16 sec
 8/19 Test #18: testTriggerIntervalMonitor ........   Passed   92.27 sec
 9/19 Test  #4: testAdcChannel ....................   Passed  108.69 sec
10/19 Test #11: testLearningFeedForward ...........   Passed  111.62 sec
11/19 Test #15: testReferencePhaseMonitor_slave ...   Passed  114.96 sec
12/19 Test #14: testReferencePhaseMonitor .........   Passed  117.04 sec
13/19 Test #17: testTableGeneration ...............   Passed  170.10 sec
14/19 Test  #8: testControllerCoefficients ........   Passed  170.89 sec
15/19 Test #10: testFirmwareSEUDetection ..........   Passed  174.66 sec
16/19 Test  #3: test_llrfctrl .....................   Passed  304.39 sec
17/19 Test  #7: testBeamLoadingCompensation .......   Passed  351.71 sec
18/19 Test #12: testOVC ...........................   Passed  392.64 sec
19/19 Test  #6: testAutomation ....................   Passed  411.95 sec

100% tests passed, 0 tests failed out of 19

Total Test time (real) = 411.96 sec
```

**Review by mhier — APPROVED** [review]

**Pull request merged**